reach2

```diff
@@ -0,0 +1,13 @@
+#!/bin/sh
+
+case "$1" in
+  "bootstrap") echo "TODO" ;; # Document this in `optparse` as well
+
+  *)
+    s=$(docker run -it --rm reachsh/reach-cli:latest "$@")
```

Comment:
I was thinking
```
if docker run .... > /tmp/reach.out ; then
 /bin/sh /tmp/reach.out
else
 cat /tmp/reach.out
fi
```

I am anxious about searching in the output, the echo command-line getting too long, and the echo command doing interpolation

Reply:
I agree with you about not liking this way but the problem is finding some reliable means of distinguishing `/bin/sh`-able script (I'm thinking of regular usage but also shell completions if we get that far) and `optparse` output.

If I were to pass `reach2 --help` to your code above it'd fail and spit a bunch of lines like:
```bash
/bin/sh: n1: some text: not found
/bin/sh: n2: something else: not found
...
```

If I instead ran `reach2 this-subcommand-does-not-exist` then it would fall through to `cat` and just print the usage info correctly.

Experimenting with several variations yesterday led me to think we're not going to get around doing something like`RUNME`, but maybe there's a better way.

Perhaps we could branch on a specific exit code instead of inspecting the output? I considered doing that but wasn't thrilled with that solution either.

Also, I think interpolation is an issue whether we stuff the output in a file or not, no?

Reply:
Ok, I think that "branch on a specific exit code" is the way to go. `RUNME` replaced with `exit 42` at the end, then inspect `$?`

Reply:
And piping to a file and then executing it is guaranteed not to have the wrapper shell involved

Reply:
```bash
#!/bin/sh

IMG='reachsh/reach-cli:latest'

run () {
  docker run -t --rm "$IMG" "$@"
}

if ! (which make docker docker-compose >/dev/null 2>&1); then
  echo "Reach relies on an installation of make, docker, and docker-compose"
  exit 1
fi

if [ "$(docker image ls -q $IMG)" = '' ]; then
  if ! (docker pull "$IMG" && run update | sed 's/\r//' | sh); then exit 1; fi
fi

run "$@" >/tmp/reach.out
case "$?" in
  42) sed 's/\r//' /tmp/reach.out | sh ;;
   *) cat /tmp/reach.out ;;
esac
```

How about that?